app/vmctl/native/client.go

```diff
@@ -89,6 +91,12 @@ func (c *Client) ImportPipe(ctx context.Context, dstURL string, pr *io.PipeReade
 	if err != nil {
 		return fmt.Errorf("cannot create import request to %q: %s", c.Addr, err)
 	}
+
+	err = parseAndSetHeaders(c.Headers, req)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Do not hide logic

```suggestion
err = parseHeaders(c.Headers)
	if err != nil {
		return err
	}
for _, header := range parsedHeaders {
		req.Header.Set(header.key, header.value)
	}
```

Reply:
Oh, interesting. I was going with the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) here, but I'll defer to you on code style for this project 👍

Reply:
If you will try to write a test for this function
```func parseAndSetHeaders(headers []string, req *http.Request) error```
I think you will see why this approach is a little bit better